lib/Lib.IntTypes.fsti

```diff
@@ -1,705 +1,817 @@
 module Lib.IntTypes
 
 open FStar.Mul
-open FStar.Math.Lemmas
+
+#push-options "--max_fuel 0 --max_ifuel 1 --z3rlimit 20"
+
+// Other instances frollow from `FStar.UInt.pow2_values` which is in
+// scope of every module depending on Lib.IntTypes
+val pow2_2: n:nat -> Lemma (pow2 2 = 4)  [SMTPat (pow2 n)]
+val pow2_3: n:nat -> Lemma (pow2 3 = 8)  [SMTPat (pow2 n)]
+val pow2_4: n:nat -> Lemma (pow2 4 = 16) [SMTPat (pow2 n)]
 
 ///
 /// Definition of machine integer base types
 ///
 
 type inttype =
-  | U1 | U8 | U16 | U32 | U64 | U128
+  | U1 | U8 | U16 | U32 | U64 | U128 | S8 | S16 | S32 | S64 | S128
```

Comment:
Supporting `S128` complicates is complicated.   
There is no code using it at the moment and any code using `S128` won't extract because there is no implementation of `int128_t` in KreMLin.
Shall we remove `S128` until we need it and it is supported in KreMLin?